mathics/core/symbols.py

```diff
@@ -128,202 +128,34 @@ def __ne__(self, other) -> bool:
 
 class BaseExpression(KeyComparable):
     """
-    This is the base class from which all other Expressions are derived from.
+    This is the base class from which all other Expressions are
+    derived from.  If you think of an Expression as tree-like, then a
+    BaseExpression is a node in the tree.
 
     This class is not complete in of itself and subclasses should adapt or fill in
     what is needed
 
     Some important subclasses: Atom and Expression.
     """
 
-    options: Any
-    pattern_sequence: bool
-    unformatted: Any
     last_evaluated: Any
     # this variable holds a function defined in mathics.core.expression that creates an expression
     create_expression: Any
 
+    # __new__ seems to be used because this object references itself.
+    # In particular:
+    #    self.unformatted = self
+    #
+    # See if there's a way to get rid of this, or ensure that this isn't causing
+    # a garbage collection problem.
     def __new__(cls, *args, **kwargs):
         self = object.__new__(cls)
         self.options = None
         self.pattern_sequence = False
-        self.unformatted = self
+        self.unformatted = self  # This may be a garbage-collection nightmare.
         self._cache = None
         return self
 
-    def clear_cache(self):
```

Comment:
Why is useful to sort the method by alphabetic order?

Reply:
When we have very long lists of things it helps me to follow things. I know one can always search for stuff, but I'll just tend to scroll when things are in alphabetic order.

There are a couple of caveats though. Some things _need_ to be in a certain order. And so when the default is alphabetic, the break in order suggests that this kind of thing is going on.

Also, when there is  _small_ list of methods, ordering by logical group is okay.

Reply:
Here I agree, that is why all code editors (which I know) sort the files alphabetically.

(This may not be necessary after drying BaseElement, but as we [I] already saw in the past, it is better to do the things step by step)